src/api/app/controllers/webui/request_controller.rb

```diff
@@ -149,6 +149,12 @@ def beta_show
     @diff_limit = params[:full_diff] ? 0 : nil
     @diff_to_superseded_id = params[:diff_to_superseded]
     @actions = @bs_request.webui_actions(filelimit: @diff_limit, tarlimit: @diff_limit, diff_to_superseded: @diff_to_superseded, diffs: false)
+    reviews = @bs_request.reviews.where(state: 'new')
+    @my_open_reviews = reviews.select { |review| review.matches_user?(User.session) }
+    @is_target_maintainer = @bs_request.is_target_maintainer?(User.session)
+    @is_author = @bs_request.creator == User.possibly_nobody.login
+    @can_add_reviews = @bs_request.state.in?([:new, :review]) && (@is_author || @is_target_maintainer || @my_open_reviews.present?)
```

Comment:
In fact those 4 lines can be pushed to `BsRequest`, you are asking for data from `BsRequest` and calculating something that `BsRequest` can do it by itself if you give it what it needs. Remember [Tell, don't ask](https://www.martinfowler.com/bliki/TellDontAsk.html)

Reply:
Those 4 lines are already there in this action, I added them in #12869. At the, I was also thinking about putting this in the `BsRequest` model, but ended up not doing it. We can definitely refactor this, but in another PR since the lines are already there.

Reply:
As discussed in our standup this morning, I created [a card in our backlog](https://trello.com/c/HMiLugdh/2652-refactor-how-we-fetch-bsrequest-reviews-and-authorize-actions-based-on-roles) to tackle this.